ads/google/a4a/utils.js

```diff
@@ -960,3 +962,50 @@ export function getAmpRuntimeTypeParameter(win) {
   const art = getBinaryTypeNumericalCode(getBinaryType(win));
   return isCdnProxy(win) && art != '0' ? art : null;
 }
+
+/**
+ * Returns the fixed size of the given element, or the fixed size of its nearest
+ * ancestor that has a fixed size, if the given element has none.
+ * @param {!Window} win
+ * @param {?Element} element
+ * @return {number} The width of the given element, or of the nearest ancestor
+ *    with a fixed size, if the given element has none.
+ */
+export function getContainerWidth(win, element) {
+  let el = element;
+  let maxDepth = 100;
+  // Find the first ancestor with a fixed size
+  while (el && maxDepth--) {
+    const layout = el.getAttribute('layout');
+    switch (layout) {
+      case Layout.FIXED:
+        return parseInt(el.getAttribute('width'), 10) || 0;
+      case Layout.RESPONSIVE:
+      case Layout.FILL:
+      case Layout.FIXED_HEIGHT:
+      case Layout.FLUID:
+        // The above layouts determine the width of the element by the
+        // containing element, or by CSS max-width property.
+        const maxWidth = parseInt(computedStyle(win, el).maxWidth, 10);
+        if (maxWidth || maxWidth == 0) {
```

Comment:
`===`?  Or is the crazy behavior of `== 0` needed?

Reply:
If !maxWidth, then maxWidth is NaN, and NaN != 0.